### PR TITLE
Add ResourceManager that will remove raw recordings that exceeds timeout

### PR DESCRIPTION
**Pull request by Karolk99**
## Acknowledging the stipulations set forth:
 - [x] I hereby confirm that a Pull Request involving updates to the Software Development Kit (SDK) has been smoothly merged, currently awaits processing, or is otherwise deemed unnecessary in this context.
 - [x] I also affirm that another Pull Request, specifically addressing updates to the documentation body (commonly referred to as 'docs'), has either been successfully incorporated, is in the process of review, or is considered superfluous under the prevailing circumstances.

### CODE REVIEW

File: lib/jellyfish/resource_manager.ex [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+      files
+      |> Enum.map(&Path.join(base_path, &1))
+      |> Enum.each(&remove_recording_if_obsolete(current_time, state.recording_timeout, &1))
+    else
+      {:error, reason} ->
+        Logger.error("Resource Manager: can't list recordings, reason: #{reason}")
+    end
+
+    Process.send_after(self(), :free_resources, state.interval * 1000)
```
Comment (Rados13) [inline lib/jellyfish/resource_manager.ex:48]: suggestion: I think we could adjust accordingly to `GenServer` tutorial https://hexdocs.pm/elixir/1.16.2/GenServer.html#module-receiving-regular-messages

File: lib/jellyfish/application.ex [hunk 2]
```diff
@@ -7,6 +7,9 @@ defmodule Jellyfish.Application do
 
   require Logger
 
+  # seconds
+  @resource_manager_opts %{interval: 600, recording_timeout: 3_600}
```
[the remark this answers is not in the document]
Comment (Karolk99) [reply lib/jellyfish/application.ex:11]: The comment above:

**Review by Rados13 — APPROVED** [review]

File: test/jellyfish/resource_manager_test.exs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  alias Jellyfish.Component.Recording
+  alias Jellyfish.ResourceManager
+
+  setup do
+    recording_path = Recording.get_base_path() |> Path.join("test")
+    report_path = Path.join(recording_path, "report.json")
+
+    File.mkdir_p!(recording_path)
+    File.touch!(report_path)
```
Comment (roznawsk) [inline test/jellyfish/resource_manager_test.exs:12]: You can set the modification time with the touch function, so we don't have to `Process.sleep` later on in the test.

File: lib/jellyfish/resource_manager.ex [hunk 4]
```diff
@@ -0,0 +1,82 @@
+defmodule Jellyfish.ResourceManager do
+  @moduledoc """
+  Module responsible for deleting outdated resources.
+  Right now it only removes outdated resources created by recording component.
+  """
+
+  use GenServer, restart: :temporary
```
Comment (roznawsk) [inline lib/jellyfish/resource_manager.ex:7]: Why we don't want to restart the Manager if it crashes?

**Pull request merged**